infra/docker/Dockerfile

```
@@ -0,0 +1,39 @@
+
+WORKDIR /app
+
+# Create non-root user for security
+RUN addgroup -g 1001 -S appgroup && \
+    adduser -u 1001 -S appuser -G appgroup
+
+COPY --from=builder /app/main .
+COPY .env.example .env
+
+# Create a script to handle environment variables
+RUN echo '#!/bin/sh\n\
```

Hadolint (2.12.0)

[info] echo may not expand escape sequences. Use printf.

(SC2028)